Lux/src/image.cpp

```diff
@@ -0,0 +1,129 @@
+// Template base class for rasterized data - used as base for image classes and vector fields
+// Handles functions common to all of these classes.
+// Supports linear interpolated sampling and mip-mapping - multiple resolutions for anti-aliasing
+
+#include "image.hpp"
+
+#define I image<T>
+#define TT template< class T >
```

Comment:
i don't think these macros add clarity

Reply:
"I" corresponds to the type that I did in the class declaration. I adopted this convention from linalg.h
"TT" I came up with myself but maybe it is a bit much

Reply:
Removed "TT" but left "I"